modules/500-upmeter/docs/internal/DEVELOPMENT_RU.md

```diff
@@ -2,29 +2,89 @@
-  - количество проверок
-  - количество успешных проверок
-  - секунд было недоступно
-- раз в 5 минут агент скидывает данные в upmeter
-- для отправки используется “wal”, так что если upmeter недоступен – данные дошлются.
+Приложение состоит из трех частей:
 
-## upmeter
+- upmeter agent (ds/upmeter-agent)
+- upmeter server (sts/upmeter)
+- smoke-mini (sts/smoke-mini-{a,b,c,d,e})
 
-- Хранит всё в базе данных SQLite, в файле `/db/dowtime.db.sqlite`.
+## Агент
 
-- Умеет принимать данные от агентов:
-  - Сразу дедуплицирует данные из 30-секундных таймслотов в 5-минутные (выбирая “лучший” результат);
-  - Складывает данные в SQLite.
+Агент измеряет доступность компонентов Deckhouse. Измерение состоит в проверке состояния объекта в
+апи кластера или проверке делает ответа на HTTP-запрос к приложению. Например, что состояние пода
+Ready или что Prometheus отвечает корректно на HTTP-запрос.
 
-- Хранит 30-секундные таймслоты только за сутки, старое удаляет из таблицы.
-- Хранит 5-минутные таймслоты постоянно, ничего не удаляет.
+Логическая единица доступности называется пробой. Проба тестирует некоторую функциональность. Проба
+состоит из одной или нескольких параллельных проверок. Например, проба `cluster-scaling` состоит из
+трех проверок, проверяющих статус подов `cloud-controller-manager`, `machine-controller-manager` и
+`bashible-apiserver`.
 
-- Умеет читать CRD `Downtime`, в этих CRD:
+Если одна из проверок выявила недоступность, компонента, статус пробы принимается `down`. Помимо
+недоступного, статус может быть `up`, если нет недоступных результатов проверок, или `uncertain`,
+если все проверки не смогли установить доступность или недоступность компонентов. Так может
+получиться, если не выполнены условия, заложенные в проверку. Например, проверки, опирающиеся на
+статус пода, будут в статусе `uncertain`, если перед тем, как проверить под, не подтвердится
+доступность аписервера.
+
+Результат запуска пробы — статус доступности той функциональности, которую проба проверяет. Пробы
+запускаются периодически с заранее заданным интервалом. Самая частая проба — `dns`, — запускается 5
+раз в секунду. Самые редкие пробы запускаются раз в минуту, например, проверка жизненного цикла
+`namespace`.
+
+Результаты проверок снимаются с минимальной необходимой частотой — 5 раз в секунду и собираются в
+массив статусов проб. Накопление статусов длится 30 секунд. Каждые 30 секунд собирается статистика
+для каждой пробы. Статистика состоит из четырех чисел:
+
+- длительность доступности (uptime)
+- длительность недоступности (downtime)
+- длительность неопределенности (uncertain)
+- оставшееся время из 30 секунд, за которое изменения не проводилось (nodata).
+
+Измерение не проводится только в том случае, если агент не запущен.
+
+Пробы объединяются в группы доступности. На эти группы выдается SLA. Статистика доступности группы
+так же вычисляется агентом во время сбора статистики проб. Статус группы вычисляется из статусов
```

Comment:
```suggestion
также вычисляется агентом во время сбора статистики проб. Статус группы вычисляется из статусов
```

Reply:
it's not "too", it is "similarly" :)

Reply:
Let's clarify. @shvgn Is it correct now?

Reply:
Yup